src/useEvent.ts

```diff
@@ -2,6 +2,12 @@ import { useLayoutEffect, useRef } from "react";
 
 type AnyFunction = (...args: any[]) => any;
 
+/**
+ * Fixes a warning when using useLayoutEffect on server
+ * https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85
+ */
+const useBrowserLayoutEffect = typeof window !== "undefined" ? useLayoutEffect : () => {};
```

Comment:
In those links you posted, I saw a lot of variance on this check ...
Some folks are checking `typeof window !== "undefined"`, some with Next.js are checking `process.browser` ...

I'd like to know if there's a better way to detect this.

Reply:
One alternative to consider: what if the "SSR fallback" was something like this:
```suggestion
const useBrowserLayoutEffect = typeof window !== "undefined" ? useLayoutEffect : (cb) => { cb() };
```
This way, in "fallback mode", we invoke the callback immediately, which sets `latestRef.current` earlier than intended?

Reply:
Sorry for all the messages on this, I guess I'm thinking out loud here ...

I'm fine with the current logic ... "if there's no `window`, then we shouldn't use `useLayoutEffect`".

I guess I'm just thinking, why do people check for `process.browser`?  I've even seen examples that check for `process.env.ENV === 'test' && typeof window !== 'undefined' && window.document && window.document.createElement` (see `rc-utils`).   

I guess we can't really fix every possible scenario, especially without example scenarios.  
The logic in this PR should eliminate some of the warnings, and should not have any negative side effects, so I'm happy with it.  Thanks!

Reply:
> Sorry for all the messages on this, I guess I'm thinking out loud here ...
> 
> I'm fine with the current logic ... "if there's no `window`, then we shouldn't use `useLayoutEffect`".
> 
> I guess I'm just thinking, why do people check for `process.browser`? I've even seen examples that check for `process.env.ENV === 'test' && typeof window !== 'undefined' && window.document && window.document.createElement` (see `rc-utils`).
> 
> I guess we can't really fix every possible scenario, especially without example scenarios. The logic in this PR should eliminate some of the warnings, and should not have any negative side effects, so I'm happy with it. Thanks!

I've tried using `const useBrowserLayoutEffect = typeof window !== "undefined" ? useLayoutEffect : () => {};` in my NextJS app and it seems to work OK, not sure about edge cases to be honest..